src/dataflow/source/kafka.rs

```diff
@@ -264,10 +312,13 @@ impl DataPlaneInfo {
         kafka_config: ClientConfig,
         consumer_activator: Arc<Mutex<SyncActivator>>,
     ) -> DataPlaneInfo {
+        let source_id = source_id.to_string();
         DataPlaneInfo {
+            source_metrics: SourceMetrics::new(&topic_name, &source_id),
+            partition_metrics: Vec::new(),
             topic_name,
             source_name,
-            source_id: source_id.to_string(),
+            source_id,
```

Comment:
this _might_ have only been necessary for the metrics?

Reply:
Indeed, I tried to remove it, but we create partition metrics on the fly if new partitions get added, so gotta keep it around!